### PR TITLE
Unmarshal github client errors correctly

### PR DESCRIPTION
**Pull request by kargakis**
Signed-off-by: Michalis Kargakis <mkargaki@redhat.com>

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ func (c *Client) request(r *request, ret interface{}) (int, error) {
 	}
 	if !okCode {
 		clientError := ClientError{}
-		if err := json.Unmarshal(b, clientError); err != nil {
+		if err := json.Unmarshal(b, &clientError); err != nil {
```
Comment (BenTheElder) [inline prow/github/client.go]: doh 🤦‍♂️

**Comment by BenTheElder** [comment]
@cjwagner probably we should bump and roll this out?

**Comment by cjwagner** [comment]
@BenTheElder  I'll do that once this merges.

**Comment by BenTheElder** [comment]
fixes: https://github.com/kubernetes/test-infra/issues/5513

**Comment by cjwagner** [comment]
Please add a short unit test to verify this fix.

**Comment by kargakis** [comment]
Does it worth it to unit test json.Unmarshal? The standard library should already handle this.

**Comment by cjwagner** [comment]
Well the problem was our use of `json.Unmarshal` not the library call itself, but this is a pretty obvious fix without much chance for regression so I guess it doesn't really need a unit test.
/lgtm

**Pull request merged**